base/reduce.jl

```diff
@@ -360,15 +360,19 @@ sum(a::AbstractArray{Bool}) = countnz(a)
 Returns the sum of all array elements, using the Kahan-Babuska-Neumaier compensated
 summation algorithm for additional accuracy.
 """
-function sum_kbn{T<:AbstractFloat}(A::AbstractArray{T})
+function sum_kbn(A::Any)
+    T = _default_eltype(typeof(A))
     c = r_promote(+, zero(T)::T)
-    if isempty(A)
+    i = start(A)
+
```

Comment:
The common style in Base is not to have empty lines inside function definitions.